qualtran/surface_code/t_factory_utils.py

```diff
@@ -0,0 +1,91 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Sequence
+
+import cirq
+import numpy as np
+
+
+def _rotation_unitary(pauli_string: str, theta: float):
+    num_qubits = len(pauli_string)
+    u = cirq.unitary(cirq.DensePauliString(pauli_string))
+    return np.cos(theta) * np.eye(1 << num_qubits) + 1j * np.sin(theta) * u
+
+
+class PauliRotationChannel(cirq.Gate):
```

Comment:
ok I was confused when you were presenting today because I thought this might be something that already exists in cirq and I was confused how you were getting the error probabilities into it.

Long story short: I think this needs a different name. Like `NoisyPauliRotation` or `OverrotatingPauliString` or `PauliRotationWithError`. 

I would assume something called `PauliRotationChannel` would do a (noiseless) pauli rotation on a density matrix. In quantum information, channels are a superset of unitary operations and does not imply that the underlying operation _isn't_ unitary

Reply:
used NoisyPauliRotation